LanguageFeatures/Inline-classes/static_analysis_member_invocation_A03_t01.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If Dm is a getter declaration with return type R then the static
+/// type of the invocation is [T1/X1 .. Ts/Xs]R.
+///
+/// @description Checks static type of a getter of an `inline` class
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/static_type_helper.dart";
+
+inline class IC1 {
+  final num id;
+  IC1(this.id);
+
+  int get asInt => id as int;
+}
+
+inline class IC2<T extends num> {
+  final T id;
+  IC2(this.id);
+
+  double get asDouble => id as double;
+}
+
+main() {
+  IC1 ic1 = IC1(42);
+  ic1.asInt.expectStaticType<Exactly<int>>();
+  ic1.id.expectStaticType<Exactly<num>>();
+
+  IC2 ic2 = IC2<double>(3.14);
```

Comment:
We could use `IC2<dynamic>` to get the extra benefit that we can see that the value of `T` is actually used in line 36, it isn't just blindly using the bound (tiny extra benefit: it also shows that it is possible to use a super-bounded type).

Reply:
Ah, of course we can't do that! :smiley:

An inline type can _never_ be super-bounded, because that would be unsound during execution of code in the instance member bodies.

I think that will be tested later on, so drop the super-bounded type here.

Using `IC2<int>` can still reveal that we're getting the type from `T`, not from the bound.